### PR TITLE
(FACT-1044) Add Linux-specific kernel resolver

### PR DESCRIPTION
**Pull request by whopper**
This commit adds a new Linux-specific kernel resolver, which
is needed to handle the slightly changed `uname` output
of Debian 8.

Also added are updates to the Debian and Ubuntu fact-based
acceptance tests for Debian 8 and Ubuntu 15.04.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
:+1:

File: acceptance/tests/facts/debian.rb [hunk 1]
```diff
@@ -7,10 +7,23 @@
 # Facts tested: os, processors, networking, identity, kernel
 #
 
-confine :to, :platform => /debian-squeeze|debian-wheezy/
+confine :to, :platform => /debian-squeeze|debian-wheezy|debian-jessie/
 
 agents.each do |agent|
-  os_version = agent['platform'] =~ /squeeze/ ? '6' : '7'
```
Comment (branan) [inline acceptance/tests/facts/debian.rb]: These elsif chains are long enough that we might want to consider data files for this fixture data (json or yaml or something)

File: acceptance/tests/facts/ubuntu.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@
 # Facts tested: os, processors, networking, identity, kernel
 #
 
-confine :to, :platform => /ubuntu-precise|ubuntu-trusty|ubuntu-utopic/
+confine :to, :platform => /ubuntu-precise|ubuntu-trusty|ubuntu-utopic|ubuntu-vivid/
 
 agents.each do |agent|
   if agent['platform'] =~ /ubuntu-precise/
```
Comment (branan) [inline acceptance/tests/facts/ubuntu.rb]: as above

**Comment by branan** [comment]
besides my thoughts on how to create more work for whopper, this looks :+1:

**Comment by whopper** [comment]
@branan heh, agreed on the fixtures. I'll open a separate ticket for that so we can get this out in time for the great package-ing tomorrow.

**Pull request merged**